backend/app/services.py

```diff
@@ -115,6 +124,64 @@ async def handle_upload_webpage(page_url: str):
     # Return error
     raise
 
+async def handle_upload_file(file: UploadFile):
+  
+  suffix = Path(file.filename).suffix.lower()
+  allowed_suffixes = {".pdf", ".txt", ".docx", ".xlsx", ".html"}
+
+  if suffix not in allowed_suffixes:
+    raise HTTPException(status_code=400, detail={"error": "Bad Request", "message": "Only pdf, txt, docx, xlsx and html files are allowed"}, file_name=file.filename)
+  
+  try:  
+    filename_clean = clean_text(file.filename)
+
+    # Create a temporary directory
+    with tempfile.TemporaryDirectory() as temp_dir:
+      temp_file_path = os.path.join(temp_dir, file.filename)
+
+      # Save the file to the temporary directory
+      with open(temp_file_path, "wb") as temp_file:
+        content = await file.read()
+        temp_file.write(content)
+        temp_file.flush()
+
+      if suffix == ".txt":
+        loader = TextLoader(temp_file_path)
+
+      elif suffix == ".pdf":
+        loader = UnstructuredPDFLoader(temp_file_path)
+
+      elif suffix == ".docx":
+        loader = UnstructuredWordDocumentLoader(temp_file_path)
+
+      elif suffix == ".html":
+        loader = UnstructuredHTMLLoader(temp_file_path)
+      
+      elif suffix == ".xlsx":
+        loader = UnstructuredExcelLoader(temp_file_path, mode = "elements")
```

Comment:
Vad sägs om att byta ut mot ett `match`-statement?

```suggestion
      match suffix:
        case ".txt":
          loader = TextLoader(temp_file_path)
      
        case ".pdf":
          loader = UnstructuredPDFLoader(temp_file_path)
      
        case ".docx":
          loader = UnstructuredWordDocumentLoader(temp_file_path)
      
        case ".html":
          loader = UnstructuredHTMLLoader(temp_file_path)
      
        case ".xlsx":
          loader = UnstructuredExcelLoader(temp_file_path, mode="elements")
```

Se: https://docs.python.org/3/tutorial/controlflow.html#match-statements